Libraries/LibWeb/HTML/StructuredSerialize.cpp

```diff
@@ -635,14 +637,16 @@ WebIDL::ExceptionOr<void> serialize_viewed_array_buffer(JS::VM& vm, Vector<u32>&
     }
 
     // 2. Let buffer be the value of value's [[ViewedArrayBuffer]] internal slot.
-    auto* buffer = view.viewed_array_buffer();
+    JS::Value buffer = view.viewed_array_buffer();
 
     // 3. Let bufferSerialized be ? StructuredSerializeInternal(buffer, forStorage, memory).
-    auto buffer_serialized = TRY(structured_serialize_internal(vm, JS::Value(buffer), for_storage, memory));
+    auto buffer_serialized = TRY(structured_serialize_internal(vm, buffer, for_storage, memory));
 
     // 4. Assert: bufferSerialized.[[Type]] is "ArrayBuffer", "ResizableArrayBuffer", "SharedArrayBuffer", or "GrowableSharedArrayBuffer".
     // NOTE: We currently only implement this for ArrayBuffer
```

Comment:
This comment is now out of date, since you've added resizable

Reply:
As far as I understand, the comment is still correct. While transferring is implemented for `ResizableArrayBuffer` in this PR, regular (non-transfer) serialization is not. 

See _FIXME_s in `Web::HTML::serialize_array_buffer`).

I'm going to resolve some of these _FIXME_s that in next PR(s).